torch/csrc/jit/interpreter.cpp

```diff
@@ -289,12 +289,17 @@ struct CodeImpl {
     // this is done with a backward scan where we mark the first time we see it
     std::unordered_set<int> seen_registers;
     auto scanUses = [&](UseList & u) {
-      listBegin(u.free_flags);
-      for(int i = 0; i < u.values.size; i++) {
+      // scan backwards because the same value may appear > once in a use list
```

Comment:
Is this bugfix exercised by the tests now?

Reply:
no -- I am going to do a separate PR that has a concise test harness for JIT so we can easily test corner cases.

Reply:
I don't fully understand this fix. This function scans an input list to every instruction in reverse, so it doesn't really matter which input marks the value to be freed later, right? Or do we empty the registers when we load inputs and that's what's causing the failure (next inputs get undefined tensors)? A bit more detailed comment would be helpful